File: 0002-add-two-numbers/0002-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *res = new ListNode(-1);
        ListNode *ans = res;
        int carry = 0;
        while(l1 && l2){
            int sum = l1->val + l2->val + carry;
            carry = sum/10;
            sum = sum%10;
            ListNode* tmp = new ListNode(sum);
            ans->next = tmp;
            ans = ans->next;
            
            l1= l1->next;
            l2= l2->next;
        }
        while(l1){
            int sum = l1->val+ carry;
            carry = sum/10;
            sum = sum%10;
            ListNode* tmp = new ListNode(sum);
            ans->next = tmp;
            ans = ans->next;
            l1= l1->next;
        }
        while(l2){
            int sum = l2->val+ carry;
            carry = sum/10;
            sum = sum%10;
            ListNode* tmp = new ListNode(sum);
            ans->next = tmp;
            ans = ans->next;
            l2= l2->next;
        }
        if(carry){
           ListNode* tmp = new ListNode(1);
            ans->next = tmp;
        }
        return res->next;
    }
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        res = ListNode(-1)
        ans = res
        carry = 0
        summ = 0

        while l1 or l2:
            if l1 is not None:
                summ += l1.val
                l1 = l1.next
            if l2 is not None:
                summ += l2.val
                l2 = l2.next
            summ += carry
            carry = summ // 10
            summ = summ % 10

            res.next = ListNode(summ)
            res = res.next
            summ = 0

        if carry > 0:
            res.next = ListNode(carry)

        return ans.next
        